js/src/logger.ts

```diff
@@ -217,6 +218,24 @@ declare global {
   var __inherited_braintrust_state: BraintrustState;
 }
 
+const REQUIRED_LOGIN_ATTRIBUTES = [
+  "appUrl",
+  "appPublicUrl",
+  "orgName",
+  "logUrl",
+] as const;
+const LOGIN_ATTRIBUTES = [
+  ...REQUIRED_LOGIN_ATTRIBUTES,
+  "loginToken",
+  "orgId",
+  "gitMetadataSettings",
+] as const;
+
+export type SerializedBraintrustState = Pick<
+  BraintrustState,
+  NonNullable<(typeof LOGIN_ATTRIBUTES)[number]>
```

Comment:
According to my IDE, this seems to make no difference to the type signature. I still see
```typescript
type SerializedBraintrustState = {
    appUrl: string | null;
    appPublicUrl: string | null;
    orgName: string | null;
    logUrl: string | null;
    loginToken: string | null;
    orgId: string | null;
    gitMetadataSettings?: GitMetadataSettings | undefined;
}
```

Which makes sense because typescript would have rejected your code for passing in possibly-null values like `this.appUrl`, which it doesn't know cannot be null if you are logged in.

If it's important to get this right, something like the following seems to work for me (and gives us the compiler errors we expect):

```
const REQUIRED_LOGIN_ATTRIBUTES = [
  "appUrl",
  "appPublicUrl",
  "orgName",
  "logUrl",
] as const;
const OPTIONAL_LOGIN_ATTRIBUTES = [
  "loginToken",
  "orgId",
  "gitMetadataSettings",
] as const;

type SerializedBraintrustState =
    & { [K in (typeof REQUIRED_LOGIN_ATTRIBUTES)[number]]: NonNullable<BraintrustState[K]> }
    & { [K in (typeof OPTIONAL_LOGIN_ATTRIBUTES)[number]]: BraintrustState[K] }
;
```